torch/distributed/distributed_c10d.py

```diff
@@ -74,8 +77,14 @@ def __new__(cls, name):
                              "on CPU tensors.")
         elif value == Backend.UNDEFINED:
             raise ValueError("Invalid backend: '{}'".format(name))
+        elif value != Backend.GLOO and value != Backend.NCCL and value != Backend.MPI:
+            value, _ = getattr(Backend, name.upper())
         return value
 
+    @classmethod
+    def register_new_backend(cls, module_name, instantiating_interface_name):
+        setattr(Backend, module_name.upper(), (module_name, instantiating_interface_name))
```

Comment:
I would change this to register the ProcessGroup subclass directly. This means you still need to look up the module and get the class, whereas this function could just register that class directly.

Reply:
I think here you meant to register the subclass ProcessGroup instantiation interface directly? I made the change according to this